infra/base-images/base-runner/bad_build_check

```diff
@@ -233,17 +297,21 @@ function check_mixed_sanitizers {
     esac
   fi
   local ASAN_CALLS=$(objdump -dC $FUZZER | egrep "${CALL_INSN}__asan" -c)
+  local DFSAN_CALLS=$(objdump -dC $FUZZER | egrep "${CALL_INSN}__dfsan" -c)
   local MSAN_CALLS=$(objdump -dC $FUZZER | egrep "${CALL_INSN}__msan" -c)
   local UBSAN_CALLS=$(objdump -dC $FUZZER | egrep "${CALL_INSN}__ubsan" -c)
 
   if [[ "$SANITIZER" = address ]]; then
-    check_asan_build $FUZZER $ASAN_CALLS $MSAN_CALLS $UBSAN_CALLS
+    check_asan_build $FUZZER $ASAN_CALLS $DFSAN_CALLS $MSAN_CALLS $UBSAN_CALLS
+    result=$?
+  elif [[ "$SANITIZER" = dataflow ]]; then
+    check_dfsan_build $FUZZER $ASAN_CALLS $DFSAN_CALLS $MSAN_CALLS $UBSAN_CALLS
     result=$?
```

Comment:
Could you do me favor by hardcoding return nonzero and seeing if check_build fails? I think it doesn't which is a problem.

Reply:
Yeah, this new build check is not active until merged and the base images re-built. Right now the check for dataflow build passes become it doesn't do anything.

Reply:
Err...that's not quite what I mean, I mean that if this script returns nonzero, that nothing happens.
Nevermind since this won't be easy for you to test.

Reply:
I test this locally by doing:

```bash
python infra/helper.py build_image --no-pull base-runner && ./bootstrap.sh zstd && python infra/helper.py check_build --engine dataflow zstd
```

where `bootstrap.sh` is:

```
#!/bin/bash
if [[ $# -eq 0 ]] ; then
  echo "Usage: $0 <project_name>"
  exit 1
fi

project_name=$1

python infra/helper.py build_image --no-pull $project_name \
&& python infra/helper.py build_fuzzers --engine dataflow --sanitizer dataflow $project_name
```

And have caught some issues, maybe that's what you meant. Fixed those in the latest commit, testing again.

Reply:
Yeah, now it works.

Reply:
Resolving this comment branch as well, please reopen if I'm still missing something important.

Reply:
Sorry I meant that (I think) this script tries to plumb nonzero exitcodes, even though those [are ignored.](https://github.com/google/oss-fuzz/blob/4d3073292ae43d99b1bd34b58e45021cb83760ae/infra/base-images/base-runner/test_all#L59), can you please leave a TODO to fix this?

Reply:
The codes are intentionally ignored in `test_all`, as we run checks in parallel and make a judgement based on the percentage of broken fuzz targets https://github.com/google/oss-fuzz/blob/4d3073292ae43d99b1bd34b58e45021cb83760ae/infra/base-images/base-runner/test_all#L85 -- added a comment in `test_all`

When `bad_build_check` is used directly, e.g. for testing an individual fuzz target https://github.com/google/oss-fuzz/blob/4d3073292ae43d99b1bd34b58e45021cb83760ae/infra/helper.py#L505

The exit code is used to decide whether the check has failed or not https://github.com/google/oss-fuzz/blob/4d3073292ae43d99b1bd34b58e45021cb83760ae/infra/helper.py#L514

So I think it's all good, unless I'm missing anything else.